common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+    DIRTY_FILES=${CONFIG_FILES_DIR}/dirty
+    UNCHECKED_FILES=${CONFIG_FILES_DIR}/unchecked
+    backup_file=${CONFIG_FILES_DIR}/${config_type}/${filename}
+    case "$config_type" in
+        "clean")
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "clean" "$filename" && return 1
+            ;;
+        "dirty")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" && return 1
+            ;;
+        "unchecked")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" && return 1
+            ;;
+        "deprecated")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" \
+            || log_warning "You want to store the file into ${UNCHECKED_FILES} directory."
+            _check_config_file "deprecated" "$filename" && return 1
+            ;;
+        *)
+            log_error "Unknown '$config_type' option in function add_config_file."
+            log_error "Allowed options are <clean,dirty,unchecked,deprecated>."
+            return 1
+            ;;
+    esac
+    mkdir -p `dirname ${backup_file}`
+    cp -a ${filename} ${backup_file}
+    return 0
+}
+
+add_kickstart_dir()
+{
+    # Add filename /root/preupgrade/kickstart directory
+    # File as first parameter has to exists in modules directory
+    # Usage:    add_to_kickstart_dir <filename> <description>
+    script_name=$1
+    kickstart_name="$KICKSTART_DIR/`basename $script_name`"
+    [ ! -f "$script_name" ] && log_error "$script_name does not exist." && exit_error
+    [ -f "$kickstart_name" ] && log_warning "`basename $script_name` already exists in $KICKSTART_DIR"
+    cp -af ${script_name} ${kickstart_name}
+}
+
+get_option()
+{
+    # Get option set up by preupgrade-assistant
+    # Available options are <migration, upgrade>
+    # Usage:    get_option option
+    # Returns:  0,1
+
+    option=$1
+    case ${option} in
+        "migration")
+            return ${MIGRATE}
+            ;;
+        "upgrade")
+            return ${UPGRADE}
+            ;;
+        *)
+            log_error "Supported options are 'migration', 'upgrade'."
+            ;;
+    esac
+
+}
+
 switch_to_content()
 {
-    cd $CURRENT_DIRECTORY
+    cd ${CURRENT_DIRECTORY}
```

Comment:
That does not really explain why any code would ever need to *leave* the directory (even using other than pushd/popd/), thus need this function.